source/core/utilities.cc

```diff
@@ -485,3 +530,18 @@ get_dimension(const std::string &file_name)
           "Lethe requires that the user explicitly specify the dimension of the problem within the parameter file. This can be achieved by adding set dimension = 2 or set dimension = 3 within the parameter file"));
     }
 }
+
+
+int
+get_max_number_of_boundary_conditions(const std::string &file_name)
+{
+  int max_number_of_boundary_conditions =
+    get_max_value_of_parameter(file_name, "number");
```

Comment:
If I understand properly we get the maximal value of `number`, so if we have a different number of boundary conditions between physics we have to be careful, since the check if not as robust?

Reply:
Yeah that's exactly it. If we have multiple physics, we will pre-declare the maximum number of boundary of all the physics. I could not figure out a way to robustly parse the number in another fashion... It's not the best, but clearly better.